Cairo Desktop/Cairo Desktop/MenuBar.xaml

```diff
@@ -1,219 +1,278 @@
 <?xml version="1.0" encoding="utf-8"?>
-<Window xmlns="http://schemas.microsoft.com/netfx/2007/xaml/presentation" xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml" xmlns:l10n="clr-namespace:CairoDesktop.Localization;assembly=CairoDesktop.Localization" xmlns:control="clr-namespace:CairoDesktop" x:Class="CairoDesktop.MenuBar"
-    x:Name="Window"
-    Title=""
-    WindowStyle="None"
-    AllowsTransparency="True"
-    Height="23"
-    Top="0"
-    Style="{StaticResource MenuBarWindowStyle}"
-    Left="0"
-    Topmost="True"
-    ResizeMode="NoResize"
-    ShowInTaskbar="False"
-    Focusable="False"
-    Initialized="OnWindowInitialized"
-    SourceInitialized="Window_SourceInitialized"
-    Closing="OnWindowClosing"
-    LocationChanged="Window_LocationChanged"
-    UseLayoutRounding="True"
-    MaxHeight="23">
+<Window xmlns="http://schemas.microsoft.com/netfx/2007/xaml/presentation"
+        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+        xmlns:l10n="clr-namespace:CairoDesktop.Localization;assembly=CairoDesktop.Localization"
+        xmlns:control="clr-namespace:CairoDesktop"
+        x:Class="CairoDesktop.MenuBar"
+        x:Name="Window"
+        Title=""
+        WindowStyle="None"
+        AllowsTransparency="True"
+        Height="23"
+        Top="0"
+        Style="{StaticResource MenuBarWindowStyle}"
+        Left="0"
+        Topmost="True"
+        ResizeMode="NoResize"
+        ShowInTaskbar="False"
+        Focusable="False"
+        Initialized="OnWindowInitialized"
+        SourceInitialized="Window_SourceInitialized"
+        Closing="OnWindowClosing"
+        LocationChanged="Window_LocationChanged"
+        UseLayoutRounding="True"
+        MaxHeight="23">
     <Window.DataContext>
-        <l10n:DisplayString/>
+        <l10n:DisplayString />
     </Window.DataContext>
     <DockPanel Style="{StaticResource MenuBarStyle}"
-        x:Name="CairoMenuBarContainer">
+               x:Name="CairoMenuBarContainer">
         <StackPanel Orientation="Horizontal"
-            VerticalAlignment="Top"
-            Height="24"
-            Width="Auto"
-            x:Name="CairoMenuBar">
+                    VerticalAlignment="Top"
+                    Height="24"
+                    Width="Auto"
+                    x:Name="CairoMenuBar">
             <Menu Style="{StaticResource CairoMenuBarMainContainerStyle}">
-                <MenuItem Style="{StaticResource CairoMenuItemCairoMenuStyle}" Name="CairoMenu">
+                <MenuItem Style="{StaticResource CairoMenuItemCairoMenuStyle}"
+                          Name="CairoMenu">
                     <MenuItem.Header>
-                        <Image Name="MenuIcon" Source="{StaticResource MenuIcon}"
-                            Width="38"
-                            Height="22"/>
+                        <Image Name="MenuIcon"
+                               Source="{StaticResource MenuIcon}"
```

Comment:
If you make this DynamicResource does it remove the need for the hack on theme change?

Reply:
We should go through all the image resources and convert them to use Cairo.xaml. Theme all the things